devito/interfaces.py

```diff
@@ -429,7 +452,20 @@ def _allocate_memory(self):
         self._full_data = self._data.view()
 
         if self.pad_time:
-            self._data = self._data[self.time_order:, :, :]
+            self._data = self._data[self.time_order:, ...]
+
+    def indexify(self):
+        """Convert base symbol and dimensions to indexed data accesses
+
```

Comment:
OK, I really don't think hijacking `indexify()` for this purpose is the way to go, because this function has a very specific use case and that is to translate out equivalents of `sympy.Function` to `sympy.Indexed`. Everything else is unintended "magic behaviour from a user's perspective.